src/intrinsics/prepack/global.js

```diff
@@ -159,19 +148,21 @@ export default function(realm: Realm): void {
         }
       }
       if (value instanceof ECMAScriptSourceFunctionValue || value instanceof AbstractValue) {
-        let functionDescriptor = new ObjectValue(realm, realm.intrinsics.ObjectPrototype);
-        functionDescriptor.$Set("funcValue", value, functionDescriptor);
-        functionDescriptor.$Set("argModelString", argModelString || realm.intrinsics.undefined, functionDescriptor);
-        // add to the todo list
-        realm.assignToGlobal(
-          t.memberExpression(
-            t.memberExpression(t.identifier("global"), t.identifier("__optimizedFunctions")),
-            t.identifier("" + additionalFunctionUid++)
-          ),
-          functionDescriptor
-        );
+        realm.optimizedFunctions.push({ value, argModel });
```

Comment:
What happens when I try to optimize the same function twice? Make it a set or map (of function to an optional ArgModel), and/or look for collisions (different `argModel`s).